fe/fe-core/src/main/java/com/starrocks/external/iceberg/ExpressionConverter.java

```diff
@@ -84,13 +84,15 @@ public static UnboundPredicate toIcebergExpression(Expr expr) {
                         ((StringLiteral) literal).getUnescapedValue());
                 break;
             }
-            case DATETIME:
             case DATE: {
                 unboundPredicate = Expressions.predicate(op, colName,
-                        ((DateLiteral) literal).getRealValue());
+                        Math.toIntExact(((DateLiteral) literal).toLocalDateTime()
+                                .toLocalDate().toEpochDay()));
                 break;
             }
-            default: break;
+            case DATETIME:
+            default:
+                break;
         }
         if (unboundPredicate == null) {
```

Comment:
This check is useless and can be removed.

Reply:
thanks @wyb PTAL